### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by yxl**
This will fix https://github.com/mozilla/j2me.js/issues/1019.
It includes  @mbebenita's docs on the new profiling tools in #989 (comment) and other miscellaneous updates related to Makefile and native methods.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,6 +142,59 @@ Modelines for JS files:
...
+timeline.leave(name?: string, details?: Object);
+```
+
+You must pair the `enter` and `leave` events but you don't necessarily need to specify arguments for `name` and `details`.
+
+The `name` argument can be any string and it specifies a event type. The timeline view will draw different types of events in different colors. It will also give you some statistics about the number of times a certain event type was seen, how long it took, etc.. 
+
+The `details` is an object, whose properties are shown when you hover over a timeline segment in the profiler view. You can specify this object when you call `timeline.enter` or when you call `timeline.leave`. Usually, you have more information when you call `leave` so that's a more convenient place to put it.
+
+The way in which you come up with event names can produce different results. In the `profilingWrapper` function, the `key` is used to specify the even type. At the moment, this is set to `MethodType[methodType]`, but you can easily set it to `implKey` which will give you a very detailed output.
```
Comment (mykmelez) [inline README.md]: Nit: even -> event.

File: README.md [hunk 2]
```diff
@@ -142,6 +142,59 @@ Modelines for JS files:
...
+Additionally, you may create your own timelines. At the moment there are 3:
+- `timeline`: VM Events like loading class files, linking, etc.
+- `methodTimeline`: Method execution.
+- `threadTimeline`: Thread scheduling.
+
+You may have to change the CSS height style of the `profileContainer` if you don't see all timelines.
+
+![Shumway's timeline viewer](https://cloud.githubusercontent.com/assets/311082/5998278/644761ec-aa7a-11e4-8149-3556b08b8c54.png)
+
+Top band is an overview of all the timelines. Second band is the `timeline`, third is the `threadTimeline` and finally the fourth is the `methodTimeline`. Us your mouse wheel to zoom in and out, pan and hover.
```
Comment (mykmelez) [inline README.md]: Nit: Us -> Use

File: README.md [hunk 3]
```diff
@@ -182,11 +235,11 @@ Java compiler will do nothing to ensure that implementation actually exists. At
 
 We use `Native` object in JS to handle creation and registration of `native` functions. See native.js
 
-    Native.create("name/of/function.(parameterTypes)returnType", jsFuncToCall, isAsync)
+    Native["name/of/function.(parameterTypes)returnType"] = jsFuncToCall;
 
 e.g.:
 
-    Native.create("java/lang/System.arraycopy.(Ljava/lang/Object;ILjava/lang/Object;II)V", function(src, srcOffset, dst, dstOffset, length) {...});
+    Native["java/lang/System.arraycopy.(Ljava/lang/Object;ILjava/lang/Object;II)V" = function(src, srcOffset, dst, dstOffset, length) {...};
```
Comment (mykmelez) [inline README.md]: In a followup, we should also add info about the new *Bindings* mechanism for defining natives.

File: README.md [hunk 4]
```diff
@@ -200,17 +253,31 @@ If raising a Java `Exception`, throw new instance of Java `Exception` class as d
...
+
+The `asyncImpl` is optional if part of the code doesn't make async calls, the method could return value as regular native method and the VM will handle it properly.
+
+e.g:
+
+    Native["java/lang/Thread.sleep.(J)V"] = function(delay) {
+        asyncImpl("V", new Promise(function(resolve, reject) {
+            window.setTimeout(resolve, delay.toNumber());
+        }));
+    };
```
Comment (mykmelez) [inline README.md]: This example comes after the paragraph that describes how `asyncImpl` is optional, so it looks like it's supposed to be an example of that, but it's actually an example of an unconditional asynchronous native. So I would switch the order of that paragraph and the example, i.e.:

>e.g:
>```
>    Native["java/lang/Thread.sleep.(J)V"] = function(delay) {
>        asyncImpl("V", new Promise(function(resolve, reject) {
>            window.setTimeout(resolve, delay.toNumber());
>        }));
>    };
>```
>The `asyncImpl` is optional if part of the code doesn't make async calls, the method could return value as regular native method and the VM will handle it properly.

I would also rephrase and extend this description a bit, to clarify what it means for a native method to be partly asynchronous and emphasize that you always need to add such methods to `yieldMap`:

> The `asyncImpl` call is optional if part of the code doesn't make async calls. The method can sometimes return a value synchronously, and the VM will handle it properly. However, if a native ever calls `asyncImpl`, even if it doesn't always do so, then you need to add the method to `yieldMap`.

File: README.md (hunk 3 above)
Comment (yxl) [inline README.md]: The followup bug is https://github.com/mozilla/j2me.js/issues/1223

File: README.md (hunk 4 above)
Comment (yxl) [inline README.md]: @mykmelez Fixed in  45c0837. I add an example of partly asynchronous native methods as well.

**Comment by mykmelez** [comment]
Nice example! I could use some newSleep. ;-)

**Pull request merged**